### PR TITLE
Documents: Add Schema to pre-render Document for GraphQl Output with Parameters 

### PR DESCRIPTION
**Pull request by dpfaffenbauer**
Ticket: #183

Query:

```graphql
{
  getDocument (id: 2) {
    ... on document_page {
      id,
      rendered (
        use_layout: false, 
        options: [{key: "blub", value: "blub"}]
        attributes: [{key: "blub", value: "blub"}]
        query: [{key: "blub", value: "blub"}]
      )
    }
  }
}
```

Result: 

```json
{
  "data": {
    "getDocument": {
      "id": "2",
      "rendered": "<!DOCTYPE html>\n<html lang=\"en\">\n<head>\n    <meta charset=\"UTF-8\">\n    <title>Example</title>\n</head>\n\n<body>\n\n<style type=\"text/css\">\n    body {\n        padding:0;\n        margin: 0;\n        font-family: \"Lucida Sans Unicode\", Arial;\n        font-size: 14px;\n    }\n\n    #site {\n        margin: 0 auto;\n        width: 600px;\n        padding: 30px 0 0 0;\n        color:#65615E;\n    }\n\n    h1, h2, h3 {\n        font-size: 18px;\n        padding: 0 0 5px 0;\n        border-bottom: 1px solid #001428;\n        margin-bottom: 5px;\n    }\n\n    h3 {\n        font-size: 14px;\n        padding: 15px 0 5px 0;\n        margin-bottom: 5px;\n        border-color: #cccccc;\n    }\n\n    img {\n        border: 0;\n    }\n\n    p {\n        padding: 0 0 5px 0;\n    }\n\n    a {\n        color: #000;\n    }\n\n    #logo {\n        text-align: center;\n        padding: 50px 0;\n    }\n\n    #logo hr {\n        display: block;\n        height: 1px;\n        overflow: hidden;\n        background: #BBB;\n        border: 0;\n        padding:0;\n        margin:30px 0 20px 0;\n    }\n\n    .claim {\n        text-transform: uppercase;\n        color:#BBB;\n    }\n\n    #site ul {\n        padding: 10px 0 10px 20px;\n        list-style: circle;\n    }\n\n    .buttons {\n        margin-bottom: 100px;\n        text-align: center;\n    }\n\n    .buttons a {\n        display: inline-block;\n        background: #6428b4;\n        color:#fff;\n        padding: 5px 10px;\n        margin-right: 10px;\n        width:40%;\n        border-radius: 2px;\n        text-decoration: none;\n    }\n\n    .buttons a:hover {\n        background: #1C8BC1;\n    }\n\n    .buttons a:last-child {\n        margin: 0;\n    }\n\n</style>\n\n\n<div id=\"site\">\n    <div id=\"logo\">\n        <a href=\"http://www.pimcore.com/\"><img src=\"/pimcore/static6/img/logo-claim-gray.svg\" style=\"width: 400px;\" /></a>\n        <hr />\n    </div>\n\n    </div>\n\n</body>\n</html>\n"
    }
  }
}
```

### CODE REVIEW

File: src/GraphQL/DocumentType/PageSnippetType.php [hunk 1]
```diff
@@ -50,5 +54,52 @@ public function build(&$config)
...
+                    'defaultValue' => []
+                ],
+                'use_layout' => [
+                    'type' => Type::boolean(),
+                    'description' => 'Disable Layout Rendering'
+                ]
+            ],
+            'resolve' => static function($value, $args) {
+                $documentId = $value['id'];
+                $document = PageSnippet::getById($documentId);
```
Comment (weisswurstkanone) [inline src/GraphQL/DocumentType/PageSnippetType.php]: do we need a permission check here ?

Comment (dpfaffenbauer) [reply src/GraphQL/DocumentType/PageSnippetType.php]: just double checked, but if you don't have permission, this does not get loaded.

Comment (weisswurstkanone) [reply src/GraphQL/DocumentType/PageSnippetType.php]: True, it is already handled here:

https://github.com/pimcore/data-hub/blob/10a31a94c23a8c567d00a4a6b265fdf7139cff3a/src/GraphQL/Resolver/QueryType.php#L172

Thanks for checking

**Comment by dpfaffenbauer** [comment]
💪

**Pull request merged**